### PR TITLE
feat: add stage2 for GCP

### PR DESCRIPTION
**Pull request by verbotenj**
Add support for stage2 for GCP

Comment out NodeExporter for now and will revisit it with turning off GKE Autopilot

Related #98 
```
GKE Autopilot constraints. The specific violations are:

Linux Capabilities:

SYS_TIME capability is not allowed. Only the following capabilities are allowed: AUDIT_WRITE, CHOWN, DAC_OVERRIDE, FOWNER, FSETID, KILL, MKNOD, NET_BIND_SERVICE, NET_RAW, SETFCAP, SETGID, SETPCAP, SETUID, SYS_CHROOT, SYS_PTRACE.
Host Namespace:

Enabling hostPID is not allowed.
Enabling host network is not allowed.
Host Ports:

Host ports are disallowed, specifically 9100 in container kube-rbac-proxy.
HostPath Volumes:

Using hostPath volumes with paths /sys and / is not allowed. Only paths under /var/log/ are permitted.
```

### CODE REVIEW

**Comment by verbotenj** [comment]
`kubectl get po -A
NAMESPACE         NAME                                                       READY   STATUS      RESTARTS       AGE
dmtr-system       alertmanager-alertmanager-0                                2/2     Running     1 (114m ago)   116m
dmtr-system       alertmanager-alertmanager-1                                2/2     Running     1 (114m ago)   116m
dmtr-system       dmtrd-77ccbfdddc-xjbb4                                     1/1     Running     0              116m
dmtr-system       grafana-0                                                  1/1     Running     0              51m
dmtr-system       kube-state-metrics-64959d64cf-2r7cp                        3/3     Running     0              115m
dmtr-system       node-exporter-2nfbw                                        2/2     Running     0              20s
dmtr-system       node-exporter-hs7lj                                        2/2     Running     0              20s
dmtr-system       node-exporter-hxxgj                                        2/2     Running     0              20s
dmtr-system       prometheus-operator-55855b44c5-22bsn                       1/1     Running     0              13h
dmtr-system       prometheus-prometheus-0                                    2/2     Running     0              51m
gke-gmp-system    alertmanager-0                                             2/2     Running     2 (13h ago)    13h
gke-gmp-system    collector-6h6dh                                            2/2     Running     2 (13h ago)    13h
gke-gmp-system    collector-7jtgs                                            2/2     Running     1 (48m ago)    50m
gke-gmp-system    collector-k8sb2                                            2/2     Running     1 (113m ago)   115m
gke-gmp-system    gmp-operator-6d5d75fc68-7bsq6                              1/1     Running     0              13h
gke-gmp-system    rule-evaluator-79f54d68fd-qpx5j                            2/2     Running     2 (13h ago)    13h
gke-managed-cim   kube-state-metrics-0                                       2/2     Running     0              13h
kube-system       anetd-57f9x                                                2/2     Running     0              50m
kube-system       anetd-8s6j5                                                2/2     Running     0              13h
kube-system       anetd-t26kn                                                2/2     Running     0              115m
kube-system       antrea-controller-horizontal-autoscaler-6dfb5849b6-jx7b5   1/1     Running     0              13h
kube-system       egress-nat-controller-5b7bfd7485-crh5g                     1/1     Running     0              13h
kube-system       event-exporter-gke-7d996c57bf-gs4ql                        2/2     Running     0              13h
kube-system       filestore-node-7rm4d                                       4/4     Running     0              115m
kube-system       filestore-node-bkq9v                                       4/4     Running     0              50m
kube-system       filestore-node-jmkw8                                       4/4     Running     0              13h
kube-system       fluentbit-gke-big-krml6                                    2/2     Running     0              13h
kube-system       fluentbit-gke-big-p94fv                                    2/2     Running     0              50m
kube-system       fluentbit-gke-big-rtwfb                                    2/2     Running     0              115m
kube-system       gcsfusecsi-node-jv8fz                                      2/2     Running     0              115m
kube-system       gcsfusecsi-node-th8j4                                      2/2     Running     0              13h
kube-system       gcsfusecsi-node-tq7v5                                      2/2     Running     0              50m
kube-system       gke-metadata-server-bs9d8                                  1/1     Running     0              50m
kube-system       gke-metadata-server-fjwdx                                  1/1     Running     0              115m
kube-system       gke-metadata-server-xs7jc                                  1/1     Running     0              13h
kube-system       gke-metrics-agent-lmb26                                    2/2     Running     0              115m
kube-system       gke-metrics-agent-mzgwb                                    2/2     Running     0              13h
kube-system       gke-metrics-agent-x7nxh                                    2/2     Running     0              50m
kube-system       image-package-extractor-9hzsl                              1/1     Running     0              13h
kube-system       image-package-extractor-cleanup-28598400-gmdbv             0/1     Completed   0              123m
kube-system       image-package-extractor-phmzz                              1/1     Running     0              49m
kube-system       image-package-extractor-xsfw7                              1/1     Running     0              114m
kube-system       ip-masq-agent-497rd                                        1/1     Running     0              13h
kube-system       ip-masq-agent-bjdn2                                        1/1     Running     0              115m
kube-system       ip-masq-agent-fcg8h                                        1/1     Running     0              50m
kube-system       konnectivity-agent-57ff985954-8589s                        2/2     Running     0              13h
kube-system       konnectivity-agent-57ff985954-wlq59                        2/2     Running     0              59m
kube-system       konnectivity-agent-57ff985954-wwsj7                        2/2     Running     0              50m
kube-system       konnectivity-agent-autoscaler-5847cf65c7-5zc2z             1/1     Running     0              13h
kube-system       kube-dns-6f955b858b-69h2m                                  4/4     Running     0              13h
kube-system       kube-dns-6f955b858b-w7rp7                                  4/4     Running     0              115m
kube-system       kube-dns-autoscaler-755c7dfdf5-pwzdm                       1/1     Running     0              13h
kube-system       l7-default-backend-6779bb6c8d-p55pn                        1/1     Running     0              13h
kube-system       metrics-server-v0.6.3-764c8d87d9-rcxsg                     2/2     Running     0              13h
kube-system       netd-6hhrv                                                 2/2     Running     0              50m
kube-system       netd-cnvzg                                                 2/2     Running     0              13h
kube-system       netd-pcn6t                                                 2/2     Running     0              115m
kube-system       node-local-dns-4v9mf                                       1/1     Running     0              115m
kube-system       node-local-dns-kdjmk                                       1/1     Running     0              13h
kube-system       node-local-dns-w4m6p                                       1/1     Running     0              50m
kube-system       pdcsi-node-5p22l                                           2/2     Running     0              50m
kube-system       pdcsi-node-lnttb                                           2/2     Running     0              13h
kube-system       pdcsi-node-lx44n                                           2/2     Running     0              115m`

File: modules/common/o11y/stage2/node-exporter.tf [hunk 1]
```diff
@@ -227,14 +219,14 @@ resource "kubernetes_daemonset" "node_exporter" {
...
-            path = "/sys"
+            path = "/var/log"
           }
           name = "sys"
         }
 
         volume {
           host_path {
-            path = "/"
+            path = "/var/log"
```
Comment (wolf31o2) [inline modules/common/o11y/stage2/node-exporter.tf:229]: Don't change this due to GKE Autopilot.

File: stage2/gcp-terraform/main.tf [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+#   source = "../modules/postgresql/stage2"
+# }
+
+# Nodeexporter deamonset requires defined resource quota as part of critical pods
+module "o11y" {
+  source        = "../../modules/common/o11y/stage2"
+  namespace     = var.dmtr_namespace
+  storage_class = "standard"
+  depends_on    = [kubernetes_resource_quota.critical_pods_quota]
+}
```
Comment (wolf31o2) [inline stage2/gcp-terraform/main.tf:41]: Comment this out for now w/ a TODO to revisit since we're using GKE Autopilot currently

**Review by wolf31o2 — CHANGES_REQUESTED** [review]
Let's leave the o11y module alone for now

**Review by wolf31o2 — APPROVED** [review]

**Pull request merged**